src/main/java/edu/cornell/kfs/fp/businessobject/PurchasingDataBase.java

```diff
@@ -0,0 +1,525 @@
+     */
+    public void setTypeOfSupply(String typeOfSupply) {
+      this.typeOfSupply = typeOfSupply;
+    }
+
+
+    /**
+     * @return the serviceIdentifier
+     */
+    public String getServiceIdentifier() {
+      return serviceIdentifier;
+    }
+
+
+    /**
+     * @param serviceIdentifier the serviceIdentifier to set
+     */
+    public void setServiceIdentifier(String serviceIdentifier) {
+      this.serviceIdentifier = serviceIdentifier;
+    }
+
+
+    /**
+     * @return the messageIdentifier
+     */
+    public String getMessageIdentifier() {
+      return messageIdentifier;
+    }
+
+
+    /**
+     * @param messageIdentifier the messageIdentifier to set
+     */
+    public void setMessageIdentifier(String messageIdentifier) {
+      this.messageIdentifier = messageIdentifier;
+    }
+
+
+    /**
+     * @return the itemSequenceNumber
+     */
+    public Integer getItemSequenceNumber() {
+      return itemSequenceNumber;
+    }
+
+
+    /**
+     * @param itemSequenceNumber the itemSequenceNumber to set
+     */
+    public void setItemSequenceNumber(Integer itemSequenceNumber) {
+      this.itemSequenceNumber = itemSequenceNumber;
+    }
+
+
+    /**
+     * @param itemSequenceNumber the itemSequenceNumber to set
+     */
+    public void setItemSequenceNumber(String itemSequenceNumber) {
+      this.itemSequenceNumber = Integer.parseInt(itemSequenceNumber);
+    }
+
+
+    /**
+     * @return the lineItemDetailIndicator
+     */
+    public Integer getLineItemDetailIndicator() {
+      return lineItemDetailIndicator;
+    }
+
+
+    /**
+     * Documentation indicates that this must be one of two values:
+     * 1=Last line item detail record
+     * 2=Other line item detail record
+     * 
+     * However, review of the file indicates that we actually observe:
+     * 0=Last line item detail record
+     * 1=Other line item detail record
+     * 
+     * @param lineItemDetailIndicator the lineItemDetailIndicator to set
+     */
+    public void setLineItemDetailIndicator(Integer lineItemDetailIndicator) {
+      this.lineItemDetailIndicator = lineItemDetailIndicator;  
+    }
+
+
+    /**
+     * @param lineItemDetailIndicator the lineItemDetailIndicator to set
+     */
+    public void setLineItemDetailIndicator(String lineItemDetailIndicator) {
+      this.setLineItemDetailIndicator(Integer.parseInt(lineItemDetailIndicator));
+    }
+    
+    /**
+     * Parses a supposed Type 50 record addendum line
+     * 
+     * @param line The current line
+     * @param lineCount The current line number
+     * @throws IOException 
+     * @throws Exception When there is a string parsing error or a missing required field
+     */
+    public void parse(String line, int lineCount) throws IOException {
+      // Lines with a * indicate data that the documentation states will be available in the future
+      this.setRecordId(USBankRecordFieldUtils.extractNormalizedString(line, 0, 2, true, lineCount));
```

Comment:
Apart from either defining lots of constants or reading the values from some file or parameter, I'm not sure of an easy way to avoid all these hard-coded amounts. These will likely be okay for now, but a more flexible approach may be needed in the future if we anticipate changes to the formatting.